src/framework/step/mod.rs

```diff
@@ -0,0 +1,132 @@
+use crate::framework::FrameworkError;
+#[cfg(feature = "enable-serde")]
+use serde::{
+    de::{Error as SerdeError, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+use std::fmt::Formatter;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Step {
+    Example(ExampleCircuit),
+    ModulusConversion(ModulusConversionDetails),
+}
+
+impl Step {
+    pub fn to_path(&self) -> String {
```

Comment:
@akoshelev this is pretty cumbersome. got any better ideas of how to serialize `Step`s?